src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ImportThunk.cs

```diff
@@ -71,5 +71,14 @@ protected override string GetName(NodeFactory factory)
         }
 
         public override int ClassCode => 433266948;
+
+        public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
+        {
+            ImportThunk otherNode = (ImportThunk)other;
+            int result = _thunkKind.CompareTo(otherNode._thunkKind);
+            if (result != 0) return result;
+
+            return comparer.Compare(_instanceCell, otherNode._instanceCell);
```

Comment:
This comparison is missing _helperCell and _moduleImport

Reply:
Those are created by factory. I assumed that we're using the same factory for the whole compilation. Is that correct?

Reply:
There's one factory yes, but that's irrelevant here. _helperCell can be different depending on the helperID used.
Regarding _moduleImport, yes you are correct, it's not needed. I missed that part.

Reply:
Can I change `NodeCache<ReadyToRunHelper, ISymbolNode> _constructedHelpers` to `NodeCache<ReadyToRunHelper, ISortableSymbolNode> _constructedHelpers` then?

Reply:
> Can I change `NodeCache<ReadyToRunHelper, ISymbolNode> _constructedHelpers` to `NodeCache<ReadyToRunHelper, ISortableSymbolNode> _constructedHelpers` then?

Why?

Reply:
I need a ISortable to pass to comparer. Right now _helperCell is just a ISymbolNode for some reason.

Reply:
You can cast _helperCell to ISortableSymbolNode

Reply:
Sure. Is there a reason the NodeCache doesn't just map ReadyToRunHelper to Imports (it seems that's all that's added). Is it that we don't want to expose all the Import functionality or that in the future there might be non-import things added to it?

Reply:
No reason. You should be able to inline the node creation and delete the `CreateReadyToRunHelperCell` API if you wish to:
``` c#
            _constructedHelpers = new NodeCache<ReadyToRunHelper, Import>(helperId =>
            {
                return new Import(EagerImports, new ReadyToRunHelperSignature(helperId));
            });
```